buildtest/cli/compilers.py

```diff
@@ -151,14 +153,18 @@ def compiler_find(configuration, modulepath=None, detailed=None, update=None):
     # print out all compilers from existing configuration file
     # run buildtest config compilers find --update to update existing configuration file
 
+    # check for edge case and raise exception when --file specifies a directory path
+    if filepath and os.path.isdir(filepath):
```

Comment:
i think you should use `is_dir(filepath)` instead of `os.path.isdir` because we also take into account variable expansion. that being said i think you may want to do a little bit more.

do the following

```
resolved_filepath = resolve_path(filepath, exist=False)

if is_dir(resolved_filepath):
  raise BuildTestError(f"The file: {resolved_filepath} is a directory, please specify a file path")
```

You may want to wrap this around a

```
if filepath:
```

I am not sure what happens when `None` is specified to `resolve_path`. 

Alternatively you can specify above your code something like this

```
dest_filepath = os.path.join(os.path.dirname(configuration.file), fname)

if filepath:
  resolved_filepath = resolve_path(filepath, exist=False)

  if is_dir(resolved_filepath):
    raise BuildTestError(f"The file: {resolved_filepath} is a directory, please specify a file path")
  dest_filepath = resolved_filepath
```